src/pages/[platform]/build-a-backend/server-side-rendering/nuxt/index.mdx

```diff
@@ -0,0 +1,626 @@
+import config from '../amplify_outputs.json';
+
+const client = generateClient<Schema>();
+
+export default defineNuxtPlugin({
+  name: 'AmplifyAPIs',
+  enforce: 'pre',
+
+  setup() {
+    // This configures Amplify on the client side of your Nuxt app
+    Amplify.configure(config, { ssr: true });
+
+    return {
+      provide: {
+        // You can add the Amplify APIs that you will use on the client side
+        // of your Nuxt app here.
+        //
+        // You can call the API by via the composable `useNuxtApp()`. For example:
+        // `useNuxtApp().$Amplify.Auth.fetchAuthSession()`
+        Amplify: {
+          Auth: {
+            fetchAuthSession,
+            fetchUserAttributes,
+            signIn,
+            signOut
+          },
+          Storage: {
+            list
+          },
+          GraphQL: {
+            client
+          }
+        }
+      }
+    };
+  }
+});
+```
+
+<Callout warning="true">
+
+Make sure you call `Amplify.configure` as early as possible in your application’s life-cycle. A missing configuration or `NoCredentials` error is thrown if `Amplify.configure` has not been called before other Amplify JavaScript APIs. Review the [Library Not Configured Troubleshooting guide](/gen1/[platform]/build-a-backend/troubleshooting/library-not-configured/) for possible causes of this issue.
+
+</Callout>
+
+### Implement `01.amplify-apis.server.ts`
+
+Example implementation:
+
+```ts title="plugins/01.amplify-apis.server.ts"
+import type { CookieRef } from 'nuxt/app';
+import {
+  createKeyValueStorageFromCookieStorageAdapter,
+  createUserPoolsTokenProvider,
+  createAWSCredentialsAndIdentityIdProvider,
+  runWithAmplifyServerContext
+} from 'aws-amplify/adapter-core';
+import { parseAmplifyConfig } from 'aws-amplify/utils';
+import {
+  fetchAuthSession,
+  fetchUserAttributes,
+  getCurrentUser
+} from 'aws-amplify/auth/server';
+import { list } from 'aws-amplify/storage/server';
+import { generateClient } from 'aws-amplify/api/server';
+import type { ListPaginateWithPathInput } from 'aws-amplify/storage';
+import type {
+  LibraryOptions,
+  FetchAuthSessionOptions
+} from '@aws-amplify/core';
+import type { Schema } from '~/amplify/data/resource';
+
+import config from '../amplify_outputs.json';
+
+// parse the content of `amplify_outputs.json` into the shape of ResourceConfig
+const amplifyConfig = parseAmplifyConfig(config);
+
+// create the Amplify used token cookies names array
+const userPoolClientId = amplifyConfig.Auth!.Cognito.userPoolClientId;
+const lastAuthUserCookieName = `CognitoIdentityServiceProvider.${userPoolClientId}.LastAuthUser`;
+
+// create a GraphQL client that can be used in a server context
+const gqlServerClient = generateClient<Schema>({ config: amplifyConfig });
+
+// Extra the model operation function types for creating wrapper function later
+type RemoveFirstParam<Params extends any[]> = Params extends [infer _, ...infer Rest] ? Rest : never; 
```

Comment:
can this be removed now in favor of `Schema["SomeModel"]["type"]`?

Reply:
The model operation APIs have a second parameter as well as `options`, `Schema["SomeModel"]["type"]` is insufficient for this use case.

Reply:
For future ref: @svidgen is working on a refactor that will expose additional CRUDL input types directly on the `Schema` type. E.g. you'll be able to call `Schema['ModelName']['createType']`. Full list [here](https://github.com/aws-amplify/amplify-api-next/pull/206/files#diff-fc8718b93a5b01e4886e15d9a14390a2fe8025a7b712bd793a23afc4b11ca151R30-R36). Let us know if this will solve your intended use case.